src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs

```diff
@@ -4,24 +4,37 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Api;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.ServiceHub.Framework;
 using Microsoft.VisualStudio.LanguageServer.Client;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LogHub;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
 using Microsoft.VisualStudio.Threading;
 using Nerdbank.Streams;
 using Roslyn.Utilities;
+using StreamJsonRpc;
 using VSShell = Microsoft.VisualStudio.Shell;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
 {
-    internal abstract class AbstractInProcLanguageClient : ILanguageClient
+    internal abstract partial class AbstractInProcLanguageClient : ILanguageClient, ILanguageServerFactory
```

Comment:
AbstractInProcLanguageClient now implements ILanguageServerFactory and passes in the jsonrpc instance during ActivateAsync

Reply:
I'm not seeing in this PR where this is used? Is this a case of the new API just lining up nicely such that no source changes were needed, or is there more changes coming?

Reply:
Ah the implementation is a bit further down in the class (the only method that wasn't copied), see `public InProcLanguageServer Create(...` - that bit is new

Reply:
Yeah, I can see the implementation, but where is it called? None of the inheritors of this type export themselves as a factory, and line 206 which calls the create method is strongly typed and doesn't need the interface.